client/my-sites/people/followers-list/followers.jsx

```diff
@@ -0,0 +1,270 @@
+			}
+			return <EmptyContent title={ emptyTitle } action={ this.renderInviteFollowersAction() } />;
+		}
+
+		let headerText;
+		if ( this.props.totalFollowers ) {
+			headerText = this.props.translate(
+				'You have %(number)d follower',
+				'You have %(number)d followers',
+				{
+					args: { number: this.props.totalFollowers },
+					count: this.props.totalFollowers,
+				}
+			);
+
+			if ( this.props.type === 'email' ) {
+				headerText = this.props.translate(
+					'You have %(number)d follower receiving updates by email',
+					'You have %(number)d followers receiving updates by email',
+					{
+						args: { number: this.props.totalFollowers },
+						count: this.props.totalFollowers,
+					}
+				);
+			}
+		}
+
+		let followers;
+		if ( this.props.followers.length ) {
+			if ( this.props.query.search && this.props.totalFollowers ) {
+				headerText = this.props.translate(
+					'%(numberPeople)d Follower Matching {{em}}"%(searchTerm)s"{{/em}}',
+					'%(numberPeople)d Followers Matching {{em}}"%(searchTerm)s"{{/em}}',
+					{
+						count: this.props.followers.length,
+						args: {
+							numberPeople: this.props.totalFollowers,
+							searchTerm: this.props.query.search,
+						},
+						components: {
+							em: <em />,
+						},
+					}
+				);
+			}
+
+			const infiniteListConditionals = {
+				fetchingNextPage: this.props.fetching,
+				lastPage: this.isLastPage(),
+			};
+
+			followers = (
+				<InfiniteList
+					{ ...infiniteListConditionals }
+					key={ key }
+					items={ this.props.followers }
+					className="followers-list__infinite is-people"
+					ref={ this.infiniteList }
+					fetchNextPage={ this.fetchNextPage }
+					getItemRef={ this.getFollowerRef }
+					renderLoadingPlaceholders={ this.renderPlaceholders }
+					renderItem={ this.renderFollower }
+					guessedItemHeight={ 126 }
+				/>
+			);
+		} else {
+			followers = this.renderPlaceholders();
+		}
+
+		const downloadListLink =
+			this.props.query.type === 'email' && !! this.props.site
+				? 'https://dashboard.wordpress.com/wp-admin/index.php?page=stats&blog=' +
+				  this.props.site.ID +
+				  '&blog_subscribers=csv&type=email'
+				: null;
+
+		return (
+			<>
+				<PeopleListSectionHeader
+					isFollower
+					isPlaceholder={ this.props.fetching || this.props.query.search }
+					label={ headerText }
+					site={ this.props.site }
+				>
+					{ downloadListLink && (
+						<Button href={ downloadListLink } compact>
+							{ this.props.translate( 'Download Data as CSV' ) }
+						</Button>
+					) }
+				</PeopleListSectionHeader>
+				<Card className={ listClass }>{ followers }</Card>
+				{ this.isLastPage() && <ListEnd /> }
+			</>
+		);
+	}
+}
+
+const mapStateToProps = ( state, ownProps ) => {
+	return {
+		followers: getFollowersForQuery( state, ownProps.query ),
+		fetching: isFetchingFollowersForQuery( state, ownProps.query ),
+		totalFollowers: getTotalFollowersForQuery( state, ownProps.query ),
+	};
+};
```

Comment:
Nit: this could be simplified to:

```suggestion
const mapStateToProps = ( state, { query } ) => ( {
	followers: getFollowersForQuery( state, query ),
	fetching: isFetchingFollowersForQuery( state, query ),
	totalFollowers: getTotalFollowersForQuery( state, query ),
} );
```